File: enode/src/main/java/org/enodeframework/eventing/IEventCommittingService.java
package org.enodeframework.eventing;

import org.enodeframework.commanding.ProcessingCommand;

public interface IEventCommittingService {
    /**
     * Commit the given aggregate's domain events to the eventstore async and publish the domain events.
     */
    void commitDomainEventAsync(EventCommittingContext eventCommittingContext);

    /**
     * Publish the given domain event stream async.
     */
    void publishDomainEventAsync(ProcessingCommand processingCommand, DomainEventStream eventStream);
}

File: enode/src/main/java/org/enodeframework/eventing/IEventCommittingService.kt
package org.enodeframework.eventing

import org.enodeframework.commanding.ProcessingCommand
import java.util.concurrent.CompletableFuture

interface IEventCommittingService {
    /**
     * Commit the given aggregate's domain events to the eventstore async and publish the domain events.
     */
    fun commitDomainEventAsync(eventCommittingContext: EventCommittingContext)

    /**
     * Publish the given domain event stream async.
     */
    fun publishDomainEventAsync(processingCommand: ProcessingCommand, eventStream: DomainEventStream): CompletableFuture<Boolean>
}